### ISSUE TITLE
[java] CloseResource Rule false positive case

### ISSUE THREAD

**Issue by 0pencoding** [pmd/pmd#2456]
**Affects PMD Version:** 
6.20

**Rule:**
CloseResource
<https://pmd.github.io/latest/pmd_rules_java_errorprone.html#closeresource>

**Description:**
If you look at the code below, it is reported as an error even though the variable 'serviceIS' was closed in the finally block. Also, Stream API doc refers to the following.

```
Streams have a BaseStream.close() method and implement AutoCloseable, but nearly all stream instances do not actually need to be closed after use. Generally, only streams whose source is an IO channel (such as those returned by Files.lines(Path, Charset)) will require closing. Most streams are backed by collections, arrays, or generating functions, which require no special resource management. (If a stream does require closing, it can be declared as a resource in a try-with-resources statement.)
```

**Code Sample demonstrating the issue:**

```java
@Override public void run() throws IOException {
  FileObject service=FileUtil.createData(getProject().getProjectDirectory(),interfaceClassPath);
  List<String> lines=new ArrayList<String>();
  InputStream serviceIS=service.getInputStream();
  try {
    BufferedReader br=new BufferedReader(new InputStreamReader(serviceIS,"UTF-8"));
    String line;
    while ((line=br.readLine()) != null) {
      lines.add(line);
    }
  }
  finally {
    serviceIS.close();
  }
  OutputStream os=service.getOutputStream();
  try {
    PrintWriter pw=new PrintWriter(new OutputStreamWriter(os,"UTF-8"));
    Iterator<String> it=lines.iterator();
    while (it.hasNext()) {
      String line=it.next();
      if (it.hasNext() || !line.trim().equals("")) {
        pw.println(line);
      }
    }
    pw.println(implClass);
    pw.flush();
  }
  finally {
    os.close();
  }
}

```

**Expected outcome:**

*   Does PMD report a violation, where there shouldn't be one? -> false-positive

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
CLI
